src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,149 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        final Node<T> newNode = new Node<>(last, value, null);
```

Comment:
Don't forget that you can have `null` as a method parameter in the `add(T value, int index)` method. This case seems to be handled correctly, as 'last' can be null when the list is empty.